js/components/Tabs.js

```diff
@@ -0,0 +1,38 @@
+const React = require("react");
+const ReactDOM = require("react-dom");
+const { bindActionCreators } = require("redux");
+const { connect } = require("react-redux");
+const { getTabs } = require("../queries");
+const actions = require("../actions")
+
+require("./tabs.css");
+const dom = React.DOM;
+
+function Tabs({tabs, selectTab, loadSources}) {
+  const tabsArr = Object.keys(tabs).map(k => tabs[k]);
+
+  function onClickTab(e) {
+    selectTab({ tabActor: e.currentTarget.dataset.actorId })
+      .then(loadSources);
```

Comment:
When you select a tab, you're doing two things: attaching to the threadClient and loading sources. Chaining these actions in the click handler seemed a little dirty, but better than other alternatives. For example, I tried to dispatch the loadSources action from within the selectTab action, but it seemed bad for Tab actions to know about Sources actions.